### PR TITLE
Add python API `pairwise_point_polygon_distance`

### LINKED ISSUE #756: [FEA] Pairwise Point Polygon Distance
**Is your feature request related to a problem? Please describe.**
Part of #231 , to support all geometry pairwise distance, point-polygon distance is needed.

**Describe the solution you'd like**
We already have point-linestring distance, we need to check if the point is contained in the polygon. If not, compute shortest distance to all rings.

### PR DESCRIPTION
**Pull request by isVoid**
## Description

This PR closes #756 , add `pairwise_point_polygon_distance` for python.

Depend on #984 #976 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cuspatial/cuspatial/core/spatial/distance.py [hunk 1]
```diff
@@ -382,6 +385,104 @@ def pairwise_point_linestring_distance(
     )
 
 
+def pairwise_point_polygon_distance(points: GeoSeries, polygons: GeoSeries):
+    """Compute distance between pairs of (multi)points and (multi)polygons
+
+    The distance between a (multi)point and a (multi)polygons
```
Comment (thomcom) [inline python/cuspatial/cuspatial/core/spatial/distance.py:391]:
```suggestion
    The distance between a (multi)point and a (multi)polygon
```

**Review by thomcom — COMMENTED** [review]
Looks overall really good, great job.

File: docs/source/user_guide/cuspatial_api_examples.ipynb [hunk 2]
```diff
@@ -1,1488 +1,1711 @@
 {
```
[the remark this answers is not in the document]
Comment (isVoid) [reply docs/source/user_guide/cuspatial_api_examples.ipynb:1]: Yeah, I'll look into this.

File: python/cuspatial/cuspatial/tests/spatial/distance/test_pairwise_point_polygon_distance.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+    expect = naturalearth_cities.geometry[:N].distance(
+        naturalearth_lowres.geometry[:N]
+    )
+
+    got = cuspatial.pairwise_point_polygon_distance(
+        gpu_cities[:N], gpu_countries[:N]
+    )
+
+    assert_series_equal(cudf.Series(expect), got)
```
[the remark this answers is not in the document]
Comment (isVoid) [reply python/cuspatial/cuspatial/tests/spatial/distance/test_pairwise_point_polygon_distance.py:142]: `distance` can not only handle point-polygon, but other geometry combination as well. I'm thinking of making `distance` when I finished `linestring-polygon` and `polygon-polygon` distance in C++. And it's tracked by: https://github.com/rapidsai/cuspatial/issues/759

File: docs/source/user_guide/cuspatial_api_examples.ipynb (hunk 2 above)
[the remark this answers is not in the document]
Comment (isVoid) [reply docs/source/user_guide/cuspatial_api_examples.ipynb]: This should be fixed now.

File: python/cuspatial/cuspatial/tests/spatial/distance/test_pairwise_point_polygon_distance.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (thomcom) [reply python/cuspatial/cuspatial/tests/spatial/distance/test_pairwise_point_polygon_distance.py:142]: :+1:

**Review by thomcom — APPROVED** [review]

**Pull request merged**